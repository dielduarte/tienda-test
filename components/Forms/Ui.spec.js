import { shallow } from 'enzyme';
import 'jest-styled-components';

import {
  Label,
  Input,
  FormRow,
  FormColumn,
  InputPrefix,
  DraftContainer
} from './Ui';

describe('Forms => Ui', () => {
  it('Label', () => {
    const wrapper = shallow(<Label />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Input', () => {
    const wrapper = shallow(<Input />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Input withError', () => {
    const wrapper = shallow(<Input withError />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Input with prefix', () => {
    const wrapper = shallow(<Input withPrefix />);
    expect(wrapper).toMatchSnapshot();
  });

  it('FormRow', () => {
    const wrapper = shallow(<FormRow />);
    expect(wrapper).toMatchSnapshot();
  });

  it('FormColumn', () => {
    const wrapper = shallow(<FormColumn />);
    expect(wrapper).toMatchSnapshot();
  });

  it('InputPrefix', () => {
    const wrapper = shallow(<InputPrefix>R$</InputPrefix>);
    expect(wrapper).toMatchSnapshot();
  });

  it('DraftContainer', () => {
    const wrapper = shallow(<DraftContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
