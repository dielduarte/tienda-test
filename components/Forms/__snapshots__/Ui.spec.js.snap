// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms => Ui DraftContainer 1`] = `
.c0 div[class^='RichTextEditor__root'],
.c0 div[class^='RichTextEditor__editor'] {
  width: 100%;
}

.c0 div[class^='RichTextEditor__editor'] {
  height: 300px;
}

<Styled(styled.div)
  className="c0"
/>
`;

exports[`Forms => Ui FormColumn 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<styled.div
  className="c0"
/>
`;

exports[`Forms => Ui FormRow 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<styled.div
  className="c0"
/>
`;

exports[`Forms => Ui Input 1`] = `
.c0 {
  border: 2px solid;
  border-radius: 5px;
  padding: 10px;
  height: 40px;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  font-size: 14px;
  margin-bottom: 20px;
}

<input
  className="c0"
/>
`;

exports[`Forms => Ui Input with prefix 1`] = `
.c0 {
  border: 2px solid;
  border-radius: 0 5px 5px 0;
  padding: 10px;
  height: 40px;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  font-size: 14px;
  margin-bottom: 20px;
}

<input
  className="c0"
/>
`;

exports[`Forms => Ui Input withError 1`] = `
.c0 {
  border: 2px solid;
  border-radius: 5px;
  padding: 10px;
  height: 40px;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  font-size: 14px;
  margin-bottom: 20px;
}

<input
  className="c0"
/>
`;

exports[`Forms => Ui InputPrefix 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 40px;
  height: 40px;
  border-radius: 5px 0 0 5px;
  margin-bottom: 20px;
}

<div
  className="c0"
>
  R$
</div>
`;

exports[`Forms => Ui Label 1`] = `
.c0 {
  font-size: 14px;
  margin-bottom: 2px;
}

<label
  className="c0"
/>
`;
